docs: Pretty print install.md (#1057)
spidernet-io/spiderpool: docs/usage/install.md

# Installation

*This guide shows how to install Spiderpool using [Helm](https://helm.sh/).* 

## Generic

> Setup the Helm repository.

Reviewer: ```suggestion
Set up the Helm repository.
```

```

More details about [Spiderpool charts parameters](https://github.com/spidernet-io/spiderpool/blob/main/charts/spiderpool/README.md#parameters).

>It should be noted that the Pods of the spiderpool-controller have to run in the hostnetwork mode, because there may be no other IPAM CNI in the cluster that can allocate some IP addresses to them.
>
>In this regard, we used `podAntiAffinity` to ensure that different replicas of spiderpool-controller will not run on the same Node to avoid host port conflicts.
>
>The replicas of spiderpool-controller can be adjusted by setting parameter `spiderpoolController.replicas`, but please ensure that you have enough Nodes to run them.

## IP version

Spiderpool can work in IPv4 only, IPv6 only or dual-stack case. For example, we can deploy Spiderpool in this way to enable dual stack:

```bash
helm install spiderpool spiderpool/spiderpool --namespace kube-system \
  --set feature.enableIPv4=true \
  --set feature.enableIPv6=true
```

By default, `feature.enableIPv4` is enabled and `feature.enableIPv6` is not.

## Certificates

Spiderpool-controller needs TLS certificates to run webhook server. We can configure it in several ways.

### Auto

> Use Helm's template function [genSignedCert](https://helm.sh/docs/chart_template_guide/function_list/#gensignedcert) to generate TLS certificates, this is the simplest and most common way to configure:

Reviewer: ```suggestion
Use Helm's template function [genSignedCert](https://helm.sh/docs/chart_template_guide/function_list/#gensignedcert) to generate TLS certificates. This is the simplest and most common way to configure:
```

```bash
helm install spiderpool spiderpool/spiderpool --namespace kube-system \
  --set spiderpoolController.tls.method=auto
```

Note that the default value of parameter `spiderpoolController.tls.method` is `auto`.

### Provided

> If you want to run spiderpool-controller with a self-signed certificate, `provided` would be a good choice. You can use OpenSSL to generate certificates, or our script:

Reviewer: ```suggestion
If you want to run spiderpool-controller with a self-signed certificate, `provided` would be a good choice. You can use OpenSSL to generate certificates, or run the following script:
```

  --set spiderpoolController.tls.provided.tlsCa=${CA} \
  --set spiderpoolController.tls.provided.tlsCert=${SERVER_CERT} \
  --set spiderpoolController.tls.provided.tlsKey=${SERVER_KEY}
```

### Cert-manager

It is **not recommended to use this mode directly**, because the Spiderpool requires the TLS certificates provided by cert-manager, while the cert-manager requires the IP address provided by Spiderpool (cycle reference).

Therefore, if possible, you must first [deploy cert-manager](https://cert-manager.io/docs/installation/) using other IPAM CNI in the cluster, and then deploy Spiderpool.

```bash
helm install spiderpool spiderpool/spiderpool --namespace kube-system \
  --set spiderpoolController.tls.method=certmanager \
  --set spiderpoolController.tls.certmanager.issuerName=${CERT_MANAGER_ISSUER_NAME}
```

## Cluster default IPPool

The cluster default IPPool usually serves some components that need IP addresses when the Kubernetes cluster is initialized, such as CoreDNS.

Prepare the IP address ranges we need to use next:

```bash
# "CIDR"
IPV4_SUBNET_YOU_EXPECT="172.18.40.0/24"
# "IP" or "IP-IP"
IPV4_IPRANGES_YOU_EXPECT="172.18.40.40-172.20.40.200"
```

> Let's create an IPv4 IPPool while deploying Spiderpool, and configure it as the cluster default IPPool in the [global Configmap configuration](https://github.com/spidernet-io/spiderpool/blob/main/docs/concepts/config.md#configmap-configuration). 

Reviewer: How about a relative url? such as `../concepts/config.md#configmap-configuration`

```

## Full example

Here is a general deployment example, which satisfies the following conditions:

- Dual stack.
- Generate TLS certificates in `auto` mode.
- Create and configure the cluster default IPv4/6 IPPool.

```bash
helm install spiderpool spiderpool/spiderpool --namespace kube-system \
  --set spiderpoolController.tls.method=auto \
  --set feature.enableIPv4=true \
  --set feature.enableIPv6=true \
  --set clusterDefaultPool.installIPv4IPPool=true \
  --set clusterDefaultPool.installIPv6IPPool=true \
  --set clusterDefaultPool.ipv4Subnet=${IPV4_SUBNET_YOU_EXPECT} \
  --set clusterDefaultPool.ipv4IPRanges={${IPV4_IPRANGES_YOU_EXPECT}} \
  --set clusterDefaultPool.ipv6Subnet=${IPV6_SUBNET_YOU_EXPECT} \
  --set clusterDefaultPool.ipv6IPRanges={${IPV6_IPRANGES_YOU_EXPECT}}
```

## Update CNI network configuration

Finally, we should edit the [CNI network configuration](https://www.cni.dev/docs/spec/#section-1-network-configuration-format) with the default path `/etc/cni/net.d` on the related Nodes so that the Main CNI can use Spiderpool IPAM CNI to allocate IP addresses. Replace the `ipam` object with the following:

```json
"ipam":{
    "type":"spiderpool"
}
```

> Take a example of Macvlan + Spiderpool:

Reviewer: ```suggestion
Take an example of Macvlan + Spiderpool:
```

```json
{
    "cniVersion":"0.4.0",
    "type":"macvlan",
    "mode":"bridge",
    "master":"eth0",
    "name":"macvlan",
    "ipam":{
        "type":"spiderpool"
    }
}
```

## Uninstall

Generally, we can uninstall Spiderpool release in this way:

```bash

> helm uninstall spiderpool -n kube-system

Reviewer 2: 可能 还有残留
